src/NSwag.Commands/Commands/Generation/OpenApiGeneratorCommandBase.cs

```diff
@@ -224,12 +224,14 @@ public async Task<TSettings> CreateSettingsAsync(AssemblyLoader.AssemblyLoader a
             JsonSerializerSettings serializerSettings;
             try
             {
-                var mvcJsonOptions = serviceProvider?.GetRequiredService<IOptions<MvcNewtonsoftJsonOptions>>();
-                serializerSettings = mvcJsonOptions?.Value?.SerializerSettings;
+                if (mvcOptions.OutputFormatters.Any(f => f.GetType() == typeof(Microsoft.AspNetCore.Mvc.Formatters.NewtonsoftJsonOutputFormatter)))
```

Comment:
Is this safe (i.e. always registered)? Also for .NET Core 1.0, etc.?
Or better: Why can we use this try/catch fallback implementation?

Reply:
The try/catch fallback relies on the assumption that System.Text.Json is the default, and will only fallback to Newtonsoft when the Newtonsoft output formatter is registered. 
Since the Microsoft.AspNetCore.Mvc.NewtonsoftJson package is referenced by this project, I don't know of any case where this will throw an exception. But there is probably some, so better safe than sorry.

This code is inside an #if NETCOREAPP3_0, so I don't think Core 1.0 support is required.